appscale/tools/agents/azure_agent.py

```diff
@@ -567,31 +567,37 @@ def terminate_instances(self, parameters):
     vmss_delete_threads = []
     deleted_instance_ids = []
     for vmss in vmss_list:
-      deleted_instance_ids.extend(
-        compute_client.virtual_machine_scale_set_vms.list(resource_group,
-                                                          vmss.name))
+      vm_list = compute_client.virtual_machine_scale_set_vms.list(
+        resource_group, vmss.name)
+      for vm in vm_list:
+        deleted_instance_ids.append(vm.name)
```

Comment:
Maybe I'm confused but are there instance ids already deleted here?
If not, please rename to _instance_ids_to_delete_.